GoASTScanner/gas has been renamed to gosec

<!--
Welcome to the Jaeger project! 👋🎉

- Please search for existing issues to avoid creating duplicate bugs/feature requests.
- Please be respectful and considerate of others when commenting on issues.
- Please provide as much information as possible so we all understand the issue.
- If you only have a question, you may get a faster response by asking in
    - our chat room https://gitter.im/jaegertracing/Lobby, or
    - the forum https://groups.google.com/d/forum/jaeger-tracing
    (but please don't double post)
-->

See https://github.com/securego/gosec/pull/216 for more details. 

This breaks our Travis builds because installing `gas` now fails

```
go get github.com/GoASTScanner/gas/cmd/gas/...
warning: "github.com/GoASTScanner/gas/cmd/gas/..." matched no packages
.
.
.
gas -quiet -exclude=G104 . ./cmd/... ./crossdock/... ./model/... ./pkg/... ./plugin/... ./storage/...
make: gas: Command not found
make: *** [lint-gas] Error 127
```

Updating our Makefiles to use `gosec` should solve this problem.

--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,7 @@ GOTEST=go test -v $(RACE)
 GOLINT=golint
 GOVET=go vet
 GOFMT=gofmt
-GAS=gas -quiet -exclude=G104
+GOSEC=gosec -quiet -exclude=G104
 FMT_LOG=fmt.log
 LINT_LOG=lint.log
 IMPORT_LOG=import.log
@@ -125,12 +125,12 @@ fmt:
 	$(GOFMT) -e -s -l -w $(ALL_SRC)
 	./scripts/updateLicenses.sh
 
-.PHONY: lint-gas
-lint-gas:
-	$(GAS) $(TOP_PKGS)
+.PHONY: lint-gosec
+lint-gosec:
+	$(GOSEC) $(TOP_PKGS)
 
 .PHONY: lint
-lint: lint-gas
+lint: lint-gosec
 	$(GOVET) $(TOP_PKGS)
 	@cat /dev/null > $(LINT_LOG)
 	$(GOLINT) $(TOP_PKGS) | \
@@ -272,7 +272,7 @@ install-tools:
 	go get golang.org/x/tools/cmd/cover
 	go get github.com/golang/lint/golint
 	go get github.com/sectioneight/md-to-godoc
-	go get github.com/GoASTScanner/gas/cmd/gas/...
+	go get github.com/securego/gosec/cmd/gosec/...
 
 .PHONY: install-ci
 install-ci: install install-tools

